scripts/deploy-transport-controller.sh

```diff
@@ -15,10 +15,66 @@
 
 # Transport deployment script assumes it runs using the kubeflex hosting cluster context.
 
-export WDS_NAME="$1" ## first argument is WDS name
-export IMBS_NAME="$2" ## second argument is IMBS name
-export TRANSPORT_CONTROLLER_IMAGE="${3:-ghcr.io/kubestellar/ocm-transport-plugin/transport-controller:0.1.0}" ## third argument is transport controller image
-export IMAGE_PULL_POLICY="${IMAGE_PULL_POLICY:=Always}"
+
+DEFAULT_TRANSPORT_CONTROLLER_IMAGE=ghcr.io/kubestellar/ocm-transport-plugin/transport-controller:0.1.3
+
+args=()
+
+TRANSPORT_CONTROLLER_IMAGE=""
+
+while [ $# != 0 ]; do
+    case "$1" in
+        (-h|--help)
+            echo "$0 usage: \$WDS_name \$ITS_name [\$transport_controller_image_ref] (--image-pull-policy \$policy | --controller-verbosity \$number | --transport-controller-image \$image_ref)*"
+            exit;;
+        (--transport-controller-image)
+          if (( $# > 1 )); then
+            TRANSPORT_CONTROLLER_IMAGE="$2"
+            shift
+          else
+            echo "Missing transport-controller-image value" >&2
+            exit 1;
+          fi;;
+        (--image-pull-policy)
+          if (( $# > 1 )); then
+            IMAGE_PULL_POLICY="$2"
+            shift
+          else
+            echo "Missing image-pull-policy value" >&2
+            exit 1;
+          fi;;
+        (--controller-verbosity)
+          if (( $# > 1 )); then
+            CONTROLLER_VERBOSITY="$2"
+            shift
+          else
+            echo "Missing controller-verbosity value" >&2
+            exit 1;
+          fi;;
+        (-*) echo "$0: unrecognized flag '$1'" >&2
+            exit 1;;
+        (*) args[${#args[*]}]="$1"
+    esac
+    shift
+done
+
+if (( ${#args[*]} < 2 || ${#args[*]} > 3 )); then
+    echo "$0: expecting two or three positional arguments" >&2
```

Comment:
I know it's more complicated to test for the exact acceptable options, but this error doesn't give much valuable info

Reply:
Yes. Because either two or three is acceptable, a decent explanation would basically recapitulate most of the usage message. I have pushded a revision that adds a reference to the usage.